### PR TITLE
rbac: Create store methods for roles and permissions (Part 1)

### PR DESCRIPTION
**Pull request by BolajiOlajide**
Part 1 #45456

## Test plan

Added unit tests for store methods

### CODE REVIEW

File: internal/database/roles.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	GetRoleOpts    RoleOpts
+)
+
+type RolesListOptions struct {
+	*LimitOffset
+}
+
+type RoleNotFoundErr struct {
+	ID int32
+}
```
Comment (eseliger) [inline internal/database/roles.go:63]: should implement `NotFound() bool`

File: internal/database/roles.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+type RoleStore interface {
+	basestore.ShareableStore
+
+	Count(ctx context.Context, opts RolesListOptions) (int, error)
+	Create(ctx context.Context, name string, readonly bool) (*types.Role, error)
+	Delete(ctx context.Context, opts DeleteRoleOpts) error
+	GetByID(ctx context.Context, opts GetRoleOpts) (*types.Role, error)
+	List(ctx context.Context, opts RolesListOptions) ([]*types.Role, error)
+	Update(ctx context.Context, role *types.Role) (*types.Role, error)
+}
```
Comment (eseliger) [inline internal/database/roles.go:46]: some doc strings would be appreciated!

File: internal/database/roles.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+		return nil, errors.Wrap(err, "scanning role")
+	}
+	return updated, nil
+}
+
+const roleDeleteQueryFmtStr = `
+UPDATE roles
+SET
+	deleted_at = NOW()
+WHERE id = %s AND readonly = false
```
Comment (eseliger) [inline internal/database/roles.go:232]:
```suggestion
WHERE id = %s AND NOT readonly
```
reads a bit nicer but super nitpicky 😬

**Review by eseliger — APPROVED** [review]

File: internal/database/roles.go (hunk 2 above)
Comment (BolajiOlajide) [reply internal/database/roles.go]: Added

File: internal/database/roles.go (hunk 1 above)
Comment (BolajiOlajide) [reply internal/database/roles.go:67]: Added.

**Review by Piszmog — APPROVED** [review]

**Pull request merged**